tests/tests/eth_provider.rs

```diff
@@ -1421,3 +1421,44 @@ async fn test_transaction_by_hash(#[future] katana_empty: Katana, _setup: ()) {
         transaction.tx
     );
 }
+
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_with_other_fields(#[future] katana: Katana, _setup: ()) {
+    let eth_provider = katana.eth_provider();
+    let transaction = katana.most_recent_transaction().unwrap();
+
+    // Retrieve receipts by block number
+    let receipts = eth_provider
+        .block_receipts(Some(alloy_rpc_types::BlockId::Number(BlockNumberOrTag::Number(
+            transaction.block_number.unwrap(),
+        ))))
+        .await
+        .unwrap()
+        .unwrap();
+    assert_eq!(receipts.len(), 1);
+    let receipt = receipts.first().unwrap();
+    assert_eq!(receipt.transaction_index, transaction.transaction_index);
+    assert_eq!(receipt.block_hash, transaction.block_hash);
+    assert_eq!(receipt.block_number, transaction.block_number);
```

Comment:
Agreed, I think that to test this properly you should:
- Add a function inside the `Katana` implementation similar to `most_recent_transaction` but something like `receipts_at(block_number)` so that you can get the receipts properly from you `Katana` test structure and then do something like:
```rust
let expected_receipts = katana.receipts_at(transaction.block_number.unwrap());
assert_eq!(receipts, expected_receipts);
```
- This avoids all of these lines and this is more precise because you can check exactly the equality.

Reply:
fixed, thanks